cupy/cuda/compiler.py

```diff
@@ -703,18 +694,30 @@ def _preprocess_hipcc(source, options):
         return re.sub('(?m)^#.*$', '', pp_src)
 
 
+def _preprocess_hiprtc(source, options):
+    # source is ignored
+    prog = _NVRTCProgram(
+        '''
+        // hiprtc segfaults if the input code is empty
+        #include <hip/hip_runtime.h>
+        __global__ void _cupy_preprocess_dummy_kernel_() { }
+        ''', '')
+    try:
+        result, _ = prog.compile(options)
+    except CompileException as e:
+        dump = _get_bool_env_variable(
+            'CUPY_DUMP_CUDA_SOURCE_ON_ERROR', False)
+        if dump:
+            e.dump(sys.stderr)
+        raise
+    assert isinstance(result, bytes)
+    return result
+
+
 _hip_extra_source = None
 
 
 def _convert_to_hip_source(source, extra_source, is_hiprtc):
-    table = [
-        ('threadIdx.', 'hipThreadIdx_'),
-        ('blockIdx.', 'hipBlockIdx_'),
-        ('blockDim.', 'hipBlockDim_'),
-        ('gridDim.', 'hipGridDim_'),
-    ]
-    for i, j in table:
-        source = source.replace(i, j)
```

Comment:
In HIP's official test suite, they've even abandoned their own quirky syntax `hipThreadIdx_x` in favor of CUDA's...